container.go

```diff
@@ -208,6 +211,29 @@ func dryInvoker(fn reflect.Value, _ []reflect.Value) []reflect.Value {
 	return results
 }
 
+type maxConcurrencyOption int
+
+// MaxConcurrency run constructors in this container with a fixed pool of executor
+// goroutines. max is the number of goroutines to start.
+func MaxConcurrency(max int) Option {
+	return maxConcurrencyOption(max)
+}
+
+func (m maxConcurrencyOption) applyOption(container *Container) {
+	container.scope.sched = &parallelScheduler{concurrency: int(m)}
+}
```

Comment:
I'm flexible on this, but how do we feel about merging the two options into one.
MaxConcurrency's behavior can be:

    0: no concurrency (default)
    >0: fixed number of workers
    -1: unbounded number of workers

If we think "negative means unlimited" is smelly, we can keep it as-is.

CC @sywhang

Reply:
That makes sense to me; I went ahead and implemented your suggestion. If max<0 proves controversial we can split it off to a separate option.